core/src/repair/quic_endpoint.rs

```diff
@@ -485,6 +486,13 @@ async fn cache_connection(
     // only by SocketAddr when establishing outgoing connections.
     let entries: [Arc<RwLock<Option<Connection>>>; 2] = {
         let mut cache = cache.write().await;
+        if cache.len() >= CONNECTION_CACHE_CAPACITY {
+            connection.close(
+                CONNECTION_CLOSE_ERROR_CODE_DROPPED,
+                CONNECTION_CLOSE_REASON_DROPPED,
+            );
+            return;
+        }
```

Comment:
This is just to limit number of connections and address https://github.com/solana-labs/solana/pull/33057#discussion_r1318937347
Follow up changes will implement a better cache eviction policy.
cc @lijunwangs